base/array.jl

```diff
@@ -27,11 +27,12 @@ import Core: arraysize, arrayset, arrayref
 
 """
     Array{T}(dims)
+    Array{T,N}(dims)
```

Comment:
Could make this `Array{T[,N]}` to be consistent with the syntax you used for `SharedArray` below